classes/test.php

```diff
@@ -411,6 +411,27 @@ public function setAssertionManager(test\assertion\manager $assertionManager = n
 			->setHandler('mockGenerator', function() use ($test) { return $test->getMockGenerator(); })
 			->setHandler('mockClass', function($class, $mockNamespace = null, $mockClass = null) use ($test) { $test->getMockGenerator()->generate($class, $mockNamespace, $mockClass); return $test; })
 			->setHandler('mockTestedClass', function($mockNamespace = null, $mockClass = null) use ($test) { $test->getMockGenerator()->generate($test->getTestedClassName(), $mockNamespace, $mockClass); return $test; })
+			->setHandler('newMockInstance', function($class, $mockNamespace = null, $mockClass = null, array $constructorArguments = null) use ($test)
+				{
+					$mockNamespace = trim($mockNamespace ?: $test->getMockGenerator()->getDefaultNamespace(), '\\');
+					$mockClass = trim($mockClass ?: $class, '\\');
+					$className = $mockNamespace . '\\' . $mockClass;
+
+					if (class_exists($className) === false)
```

Comment:
we have to check if the class exists before generating it as the mock generator does not check that for us